app/controllers/OAuth.scala

```diff
@@ -74,9 +73,9 @@ final class OAuth(env: Env, apiC: => Api) extends LilaController(env):
 
   def tokenApply = AnonBodyOf(parse.form(accessTokenRequestForm)):
     _.prepare match
-      case Validated.Valid(prepared) =>
+      case Right(prepared) =>
```

Comment:
I actually don't understand how this works? It seems the `Right` case returns `Future[Result]` but the `Left` case returns `Result`.

Reply:
Is it this?

https://github.com/lichess-org/lila/blob/26a5ccb2315c4972e7980ee4c466275cee145f13/app/http/ResponseBuilder.scala#L23

Reply:
ah, definitely it!